File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    stack<int> bag;
    int answer = 0;

    for (int i : moves) {
        i -= 1;
        for (int j = 0; j < board.size(); j++) {
            if (board[j][i] > 0) {
                int value = board[j][i];
                board[j][i] = 0;

                if (bag.size() > 0) {
                    if (bag.top() == value) {
                        answer += 2;
                        bag.pop();
                    }
                    else{
                        bag.push(value);
                    }
                }
                else{
                    bag.push(value);
                }
                break;
            }
        }
    }

    return answer;
}

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    bag = []
    answer=0
    for i in moves:
        i -=1
        for j in range(len(board)):
            if board[j][i]>0:
                bag.append(board[j][i])
                board[j][i]=0
                
                if len(bag)>1:
                    if bag[-2]==bag[-1]:
                        answer+=2
                        bag.pop()
                        bag.pop()
                break
        
    return answer